api/app/routers/snow.py

```diff
@@ -0,0 +1,30 @@
+""" Routers for Snow related data
+"""
+
+import logging
+from datetime import date, datetime
+from fastapi import APIRouter, Depends
+from app.auth import authentication_required, audit
+from app.db.crud.snow import get_most_recent_processed_snow_by_date
+from app.db.database import get_async_read_session_scope
+from app.schemas.snow import ProcessedSnowModel, ProcessedSnowResponse
+from app.utils.time import vancouver_tz
+
+logger = logging.getLogger(__name__)
+
+router = APIRouter(
+    prefix="/snow",
+    dependencies=[Depends(authentication_required), Depends(audit)],
+)
+
+
+@router.get('/most-recent-by-date/{for_date}', response_model=ProcessedSnowResponse | None)
+async def get_most_recent_by_date(for_date: date, _=Depends(authentication_required)):
+    """ Returns the most recent processed snow record before or equal to the provided date. """
+    logger.info('/snow/most-recent-by-date/')
+    tz_aware_datetime = vancouver_tz.localize(datetime.combine(for_date, datetime.min.time()))
+    async with get_async_read_session_scope() as session:
+        result = await get_most_recent_processed_snow_by_date(session, tz_aware_datetime)
+        if result is not None:
+            processed_snow = result[0]
```

Comment:
I think the query returns the `first` value from the possible list of results, will this index access work?

Reply:
The query actually returns the `first` value as a tuple, so the indexing is necessary.